signxml/__init__.py

```diff
@@ -591,9 +591,17 @@ def _apply_transforms(self, payload, transforms_node, signature, c14n_algorithm)
 
         return payload
 
-    def verify(self, data, require_x509=True, x509_cert=None, cert_subject_name=None, ca_pem_file=None, ca_path=None,
-               hmac_key=None, validate_schema=True, parser=None, uri_resolver=None, id_attribute=None,
-               expect_references=1):
+    def _get_serial_number(self, comps, serial):
+        for v in comps:
+            if len(v) == 2 and \
+               v[0].decode('utf-8') in 'serialNumber' and \
```

Comment:
This comparison seems wrong. Why are you testing for a substring of 'serialNumber'?

Reply:
:smile: definitely some autocomplete/paste issue.

I ended up omitting the serial number validation in my project and left the merge request half done, but since you want to give this attention I'll get back to this.

Thanks for a great library!